helix-core/src/main/java/org/apache/helix/controller/stages/StateTransitionThrottleController.java

```diff
@@ -208,13 +210,20 @@ protected void chargeResource(StateTransitionThrottleConfig.RebalanceType rebala
    */
   protected void chargeInstance(StateTransitionThrottleConfig.RebalanceType rebalanceType,
       String instance) {
-    if (_pendingTransitionAllowedPerInstance.containsKey(instance)
-        && _pendingTransitionAllowedPerInstance.get(instance).containsKey(rebalanceType)) {
-      chargeANYType(_pendingTransitionAllowedPerInstance.get(instance));
-      Long instanceThrottle = _pendingTransitionAllowedPerInstance.get(instance).get(rebalanceType);
-      if (instanceThrottle > 0) {
-        _pendingTransitionAllowedPerInstance.get(instance).put(rebalanceType, instanceThrottle - 1);
-      }
+    charge(rebalanceType, _pendingTransitionAllowedPerInstance.getOrDefault(instance, new HashMap<>()));
+  }
+
+  private void charge(StateTransitionThrottleConfig.RebalanceType rebalanceType,
+      Map<StateTransitionThrottleConfig.RebalanceType, Long> quota) {
+    if (StateTransitionThrottleConfig.RebalanceType.NONE.equals(rebalanceType)) {
+      logger.error("Wrong rebalance type NONE as parameter");
+      return;
+    }
+    // if ANY type is present, decrement one else do nothing
+    quota.computeIfPresent(StateTransitionThrottleConfig.RebalanceType.ANY, (key, val) -> Math.max(0, val - 1));
```

Comment:
The names are taken by method parameters. They're more important and names of (key, val) should be fine under simple logics for people familiar with the method.

Reply:
Sorry, I don't quite understand. Does `(rebalanceType, quota) -> Math.max(0, quota - 1)` not work? Are you saying it's not important to make code self-descriptive? The whole point is to make it so that it's easier to understand for people who are not familiar with what we're doing here?

Reply:
Look at the method parameter, these names are already taken

Reply:
I see what you were saying earlier. You could do `(type, quotaCount)` or something similar. Point is to make the code a little more readable :)